src/main/java/io/jenkins/plugins/oidc_provider/IdTokenCredentials.java

```diff
@@ -207,7 +208,9 @@ RSAPublicKey publicKey() {
                 } else if (t.name.equals(Claims.SUBJECT)) {
                     definedSub.set(true);
                 }
-                builder.claim(t.name, t.type.parse(Util.replaceMacro(t.format, env)));
+                if (t.requiredEnvVars == null || (t.requiredEnvVars.isEmpty() || env.keySet().containsAll(Arrays.asList(t.requiredEnvVars.split("\\s*\\s"))))) {
```

Comment:
Not following the condition. Could you explain it? Because it seems to me it should the the opposite (although I might well be misunderstanding the PR)

Reply:
But why are you claiming if the env is null or empty? shouldn't be the opposite?

Reply:
So the reason I have the requiredEnvVars as nullable is because if its missing from the JCASC config the field is set as null (this was what I noticed from the JCASC unit tests):
```
security:
  idToken:
    claimTemplates:
    - name: ok
      format: "true"
      type: boolean
```
But when its set via the gui its ""

Using a nullable string feels horrible

### Original Requirements

> Your config could be something like
> 1. sub=job:${GITHUB_REPO}:pull_request (GITHUB_REPO, CHANGE_ID)
> 2. sub=job:${GITHUB_REPO}:tag:${TAG_NAME} (GITHUB_REPO, TAG_NAME)
> 3. sub=job:${GITHUB_REPO}:tag:${BRANCH_NAME} (GITHUB_REPO, BRANCH_NAME)
> 4. sub=job:${JOB_NAME}:other (JOB_NAME)
> 5. sub=${JENKINS_URL} ()

But the main reason for the null/empty checks to allow the claim to be set is to support the final fallback condition (5) from the list above. Given what happens with JCASC/UI config this can be either `null` or `""` or if our keys match.

My original attempt at this was using `List<String> requiredEnvVars` which made everything easy as the list was either empty of not. But it kept erroring with error below that I was unable to solve:
```
java.lang.AssertionError: class java.lang.String is missing its descriptor in public final java.util.List io.jenkins.plugins.oidc_provider.config.ClaimTemplate.requiredEnvVars. See https://www.jenkins.io/redirect/developer/class-is-missing-descriptor 
```

Reply:
> when its set via the gui its ""

Use `Util.fixEmpty` on `String`-valued `@DataBoundSetter`’s.

Written for Pipeline steps but broadly applicable: https://www.jenkins.io/doc/developer/plugin-development/pipeline-integration/#handling-default-values

> My original attempt at this was using `List<String> requiredEnvVars`

Stapler/Jenkins data binding does not support `List<String>`. A simple space-separated `"VAR1 VAR2"` suffices anyway. (If you really need a list of objects you have to make the element type be `Describable`, each with its own `config.jelly`, which seems like overkill here.)

Reply:
thanks @jglick, `Utils.fixEmpty` sorted it all with the `@DataBoundSetter`